### ISSUE TITLE
✅ 🐞 Bug Report: Theme Builder broken after setting a CSS ID to an item of the Mega Menu

### ISSUE THREAD

**Issue by valallard** [elementor/elementor#21934]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Elementor's Theme Builder Edit Mode is not working anymore after setting a css ID to a specific item of the Mega Menu.
The only way to repair and have access to Edit Mode is to remove css class in Json file and upload template part again. 

NB : Edit mode is not working ONLY for the template parts that has the mega menu and the css class applied to an item.
You can still edit content like pages or whatever. 

Thank you !

### Steps to reproduce

1) Create a header in Theme Builder
2) Add a the Mega Menu widget
3) Assign an ID to an element
4) Save and close Theme Builder
5) Go back to Edit the header 
6) The header disapear and loading is infinite

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>System Info</summary>

````txt



== Server Environment ==
	Operating System: Linux
	Software: LiteSpeed
	MySQL version: MariaDB Server v10.6.11
	PHP Version: 7.4.33
	PHP Memory Limit: 512M
	PHP Max Input Vars: 5000
	PHP Max Post Size: 512M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.2
	Site URL: https://totem.tand.re
	Home URL: https://totem.tand.re
	WP Multisite: No
	Max Upload Size: 512 Mo
	Memory limit: 40M
	Max Memory limit: 512M
	Permalink Structure: /%postname%/
	Language: fr-FR
	Timezone: Europe/Paris
	Admin Email: allard.val@gmail.com
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 2.7.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: fr_FR
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36

== Active Plugins ==
	Elementor
		Version: 3.12.1
		Author: Elementor.com

	Elementor Pro
		Version: 3.12.1
		Author: Elementor.com


== Must-Use Plugins ==
	Elementor Safe Mode
		Version: 1.0.0
		Author: Elementor.com


== Utilisation des éléments ==
	
	footer : 1
		button : 1
		container : 33
		heading : 8
		icon-list : 12
		mega-menu : 1
		social-icons : 1
		spacer : 1
		text-editor : 9
		theme-site-logo : 1
	header : 2
		container : 20
		icon-list : 5
		mega-menu : 2
		spacer : 1
		text-editor : 5
		theme-site-logo : 1
	page : 1
		container : 1
		heading : 1
	wp-page : 8
		button : 1
		container : 51
		heading : 10
		icon-box : 3
		icon-list : 12
		image : 1
		text-editor : 8


== Réglages ==
	
	cpt_support: post, page
	disable_color_schemes: yes
	disable_typography_schemes: yes
	allow_tracking: yes
	css_print_method: internal
	unfiltered_files_upload: 1
	font_display: swap


== Fonctionnalités ==
	Custom Fonts: 1
	Custom Icons: 2

== Intégrations ==
	


== Expériences d’Elementor ==
	Sortie DOM optimisée: Actif par défaut
	Amélioration du chargement des ressources: Actif par défaut
	Optimisation du chargement du CSS: Actif par défaut
	Icônes de police « inline »: Inactif par défaut
	Améliorations de l’accessibilité: Actif par défaut
	Points d’arrêt personnalisés supplémentaires: Actif par défaut
	admin_menu_rearrangement: Inactif par défaut
	Conteneur Flexbox: Actif
	Mettre à niveau la bibliothèque Swiper: Actif par défaut
	Par défaut vers le Nouveau constructeur de thème: Actif par défaut
	En-tête et pied de page du thème Hello: Actif par défaut
	Editor Top Bar: Actif
	Page d’atterrissage: Actif par défaut
	Éléments imbriqués: Actif
	Chargement différé des images d’arrière-plan: Actif
	Transitions des pages: Actif par défaut
	Notes: Actif par défaut
	Boucle: Actif par défaut
	Envois de formulaire: Actif par défaut
	Accroche de défilement: Actif par défaut
	Menu: Actif


== Journal ==
	
Log: showing 20 of 362023-03-27 21:05:57 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.12.0-beta2',
  'to' => '3.12.0-beta3',
)]
2023-03-31 01:04:42 [info] elementor::elementor_updater Started 
2023-03-31 01:04:42 [info] Elementor/Upgrades - _on_each_version Start  
2023-03-31 01:04:42 [info] Elementor/Upgrades - _on_each_version Finished 
2023-03-31 01:04:42 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.12.0-beta3',
  'to' => '3.12.0',
)]
2023-03-31 01:04:42 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.12.0-beta3',
  'to' => '3.12.0',
)]
2023-03-31 01:11:08 [info] elementor-pro::elementor_pro_updater Started 
2023-03-31 01:11:08 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-03-31 01:11:08 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-03-31 01:11:08 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.12.0-beta3',
  'to' => '3.12.0',
)]
2023-04-03 22:57:10 [info] elementor::elementor_updater Started 
2023-04-03 22:57:10 [info] Elementor/Upgrades - _on_each_version Start  
2023-04-03 22:57:10 [info] Elementor/Upgrades - _on_each_version Finished 
2023-04-03 22:57:10 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.12.0',
  'to' => '3.12.1',
)]
2023-04-03 22:57:10 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.12.0',
  'to' => '3.12.1',
)]
2023-04-05 02:32:48 [info] elementor-pro::elementor_pro_updater Started 
2023-04-05 02:32:48 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-04-05 02:32:48 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-04-05 02:32:48 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.12.0',
  'to' => '3.12.1',
)]
2023-04-05 02:32:48 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.12.0',
  'to' => '3.12.1',
)]

PHP: showing 3 of 3PHP: 2023-03-26 01:20:04 [error X 6][/home/u815615354/domains/tand.re/public_html/totem/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::295] Uncaught Error: Call to a member function update_meta() on null in /home/u815615354/domains/tand.re/public_html/totem/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php:295
Stack trace:
#0 /home/u815615354/domains/tand.re/public_html/totem/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php(168): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager-&gt;save_conditions(7, Array)
#1 [internal function]: ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager-&gt;ajax_save_theme_template_conditions(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#2 /home/u815615354/domains/tand.re/public_html/totem/wp-content/plugins/elementor/core/common/modules/ajax/module.php(175): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#3 /home/u815615354/domains/tand.re/public_html/totem/wp-includes/class-wp-hook.php(308): Elementor\Core\Common\Modules\Ajax\Module-&gt;handle_ajax_request(&#039;&#039;)
#4 /home/u815615354/domains [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-04-02 01:02:47 [error X 1][/home/u815615354/domains/tand.re/public_html/totem/wp-content/plugins/elementor/core/wp-api.php::26] Uncaught Error: Call to undefined function Elementor\Core\get_plugins() in /home/u815615354/domains/tand.re/public_html/totem/wp-content/plugins/elementor/core/wp-api.php:26
Stack trace:
#0 /home/u815615354/domains/tand.re/public_html/totem/wp-content/plugins/elementor/modules/compatibility-tag/base-module.php(96): Elementor\Core\Wp_Api-&gt;get_plugins()
#1 /home/u815615354/domains/tand.re/public_html/totem/wp-content/plugins/elementor/modules/compatibility-tag/base-module.php(58): Elementor\Modules\CompatibilityTag\Base_Module-&gt;get_plugins_with_header()
#2 /home/u815615354/domains/tand.re/public_html/totem/wp-content/plugins/elementor/modules/compatibility-tag/module.php(57): Elementor\Modules\CompatibilityTag\Base_Module-&gt;get_plugins_to_check()
#3 /home/u815615354/domains/tand.re/public_html/totem/wp-content/plugins/elementor/modules/compatibility-tag/base-module.php(148): Elementor\Modules\CompatibilityTag\Module-&gt;get_plugins_to_check()
#4 /home/u815615354/domains/tand.re/public_html/totem/wp-includes/class-w [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-04-05 02:39:06 [notice X 7][/home/u815615354/domains/tand.re/public_html/totem/wp-content/plugins/elementor/core/common/modules/ajax/module.php::175] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 3 of 3JS: 2023-03-26 00:49:35 [error X 68][https://totem.tand.re/wp-admin/post.php?post=36&amp;action=elementor#:68:18] Assignment to constant variable. 
JS: 2023-03-27 23:15:24 [error X 1][https://totem.tand.re/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.12.0-beta3:3:673293] this.model.isValidChild is not a function 
JS: 2023-04-03 22:22:41 [error X 6][https://totem.tand.re/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.12.1:3:785745] Cannot read properties of undefined (reading &#039;localized&#039;) 



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Non compatible

== Elementor Pro - Compatibility Tag ==

````

</details>

**Comment by harwellian** [comment]
I can verify that this issue is happening. I emailed them through support on their website about it as well.

**Comment by jkulp** [comment]
I just had this happen on my site. Elementor Version: 3.12.1. You can regain access to the menu can be edited once you remove the IDs from the database.

**Comment by TowyTowy** [comment]
I have this issue too.

**Comment by lensbreak** [comment]
Same here.

**Comment by WebRat45** [comment]
Same here. Hope it gets fixed soon.

**Comment by cirkut** [comment]
Same issue here. I'm super glad that you helped narrow it down to adding a CSS ID! Makes it easy for the problem site I've been working on!

**Comment by JeffreyHester** [comment]
Same Here. Waiting on a fix.

**Comment by cirkut** [comment]
Appears to be fixed in [3.13's Beta](https://github.com/orgs/elementor/discussions/22119)!

**Comment by nicholaszein** [comment]
Hello everyone!

We are happy to announce that this issue has been resolved in the latest v3.13-Beta release!

We would love to see you try this new feature and give your feedback in the [Beta Announcement](https://github.com/orgs/elementor/discussions/22119)! 💪

Please try it out and let us know!

https://github.com/orgs/elementor/discussions/22119

Kind regards

PS: If you don't have time to test this out, please consider disabling the Top Bar feature until we release version 3.13.

**Comment by nicholaszein** [comment]
Hey everyone!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro **v3.12.3**! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

Cheers 🥂
